check-windows-eventlog/check-windows-eventlog.go

```diff
@@ -28,11 +28,25 @@ const (
 	errorInvalidParameter = syscall.Errno(87)
 )
 
+var (
+	rid1 = regexp.MustCompile(`^([0-9]+)$`)
+	rid2 = regexp.MustCompile(`^([0-9]+)-([0-9]+)$`)
+)
+
+type idRange struct {
+	hi   uint32
+	lo   uint32
+	bang bool
+}
+
 type logOpts struct {
 	Log            string `long:"log" description:"Event Names (comma separated)"`
 	Type           string `long:"type" description:"Event Types (comma separated)"`
 	SourcePattern  string `long:"source-pattern" description:"Event Source (regexp pattern)"`
+	SourceExclude  string `long:"source-exclude" description:"Event Source excluded (regexp pattern)"`
 	MessagePattern string `long:"message-pattern" description:"Message Pattern (regexp pattern)"`
+	MessageExclude string `long:"message-exclude" description:"Message Pattern excluded (regexp pattern)"`
+	EventID        string `long:"event-id" description:"Event IDs (separated by comma)"`
```

Comment:
Not necessary, but it's better if range specification like `${id1}-${id2}` and excluding by `!${id}` are described.